### PR TITLE
#525 Ensure invitations display when they should

### PR DESCRIPTION
**Pull request by warrensearle**
@tomlovesgithub 

Have covered the following requirement from #525
> filter out accepted and rejected invitations

We still need to display the vacancy title and only show the invite if vacancy is published

### CODE REVIEW

File: src/store/invitations.js [hunk 1]
```diff
@@ -14,17 +13,14 @@ export default {
   actions: {
     bind: firestoreAction(({ bindFirestoreRef, rootState }) => {
       if (rootState.auth.currentUser){
-        const firestoreRef = collection.where('candidate.email', '==', rootState.auth.currentUser.email);
+        const firestoreRef = collection
+          .where('candidate.email', '==', rootState.auth.currentUser.email)
+          .where('status', 'in', ['created', 'invited']);
         return bindFirestoreRef('records', firestoreRef, { serialize: vuexfireSerialize });
```
Comment (lloback) [inline src/store/invitations.js:19]: I worked with @tomlovesgithub on this today ... it is still a work in progress, but I will let him know that he needs to raise a PR urgently for his changes.

File: src/App.vue [hunk 2]
```diff
@@ -57,19 +55,49 @@ export default {
...
+      handler(val) { 
+        if (val) {
+          this.$store.dispatch('invitations/bind');
+        } else {
+          this.$store.dispatch('invitations/unbind');
+        }
+      },
+    },
+    invitations: {
+      immediate: true,
```
Comment (warrensearle) [inline src/App.vue:94]: Nice work on the watchers @tomlovesgithub 👍

**Review by warrensearle — COMMENTED** [review]
Hey @tomlovesgithub  this looks good 👍 
There are a couple of refinements we should consider:
- Before calling `vacancies/bind` we should check whether the vacancies store already has vacancies (so we don't bind twice)
- When there are invitations and the vacancies store is empty we should call `vacancies/bind` with the specific vacancy ids we want rather than loading every vacancy.

File: src/App.vue [hunk 3]
```diff
@@ -95,9 +95,11 @@ export default {
       deep: true,
       handler(val) { 
         if (val && val.length) {
-          this.$store.dispatch('vacancies/bind');
+          // if (this.$store.state.vacancies.records.length) { 
+          this.$store.dispatch('vacancies/bind', { vacancyIds: this.invitations.map(invite => invite.vacancy.id) });
+          // }
```
Comment (lloback) [inline src/App.vue:100]: don't forget to remove the commented lines

**Review by lloback — APPROVED** [review]

**Comment by warrensearle** [comment]
@lloback, @mbrookeswebdev This needs checking over and also, ideally, some tests (or test scripts) created so we can protect against regression

**Review by mbrookeswebdev — APPROVED** [review]

**Pull request merged**